pulsar-common/src/main/java/org/apache/pulsar/common/api/raw/MessageParser.java

```diff
@@ -74,7 +74,7 @@ public static void parseMessage(TopicName topicName, long ledgerId, long entryId
                 return;
             }
 
-            refCntMsgMetadata = ReferenceCountedMessageMetadata.get();
+            refCntMsgMetadata = ReferenceCountedMessageMetadata.get(headersAndPayload);
```

Comment:
hello,i wonder if this can only retain buffer for headers only. it seems `ReferenceCountedMessageMetadata` may make buffer live a bit longer which may contain the payload.